packages/@glimmer/manager/lib/internal/defaults.ts

```diff
@@ -0,0 +1,37 @@
+import { buildCapabilities } from '../util/capabilities';
+
+import type { CapturedArguments as Arguments, HelperCapabilities } from '@glimmer/interfaces';
+
+type FnArgs<Args extends Arguments = Arguments> =
+  | [...Args['positional'], Args['named']]
+  | [...Args['positional'], {}];
+
+interface FunctionHelperState<Args extends Arguments = Arguments> {
+  fn: <Return>(...args: FnArgs<Args>) => Return;
+  args: Args;
+}
+
+export class FunctionHelperManager<State extends FunctionHelperState> {
+  capabilities = buildCapabilities({
+    hasValue: true,
+    hasDestroyable: false,
+    hasScheduledEffect: false,
+  }) as HelperCapabilities;
+
+  createHelper(fn: State['fn'], args: State['args']) {
+    return { fn, args };
+  }
+
+  getValue({ fn, args }: State) {
+    let argsForFn: FnArgs<Arguments> = [
+      ...args.positional,
+      Object.keys(args.named).length > 0 ? args.named : {},
+    ];
+
+    return fn(...argsForFn);
+  }
+
+  getDebugName(fn: State['fn']) {
+    return fn.name || '(anonymous helper function)';
```

Comment:
Does anything in the build compression or magnification perhaps make `name` not what we think it is?

Reply:
Minification could change the function name. (If in module space) But folks shouldn't rely on function name stability?

Reply:
Ie doesn't have function names, it seems.

Can we drop support for ie?

Reply:
We probably shouldn't try to use `fn.name` at all, or at least not by itself, because of minification precisely (so it'll probably just be `"zw"` or something cryptic in prod). We could do something like:

```js
if (fn.name) {
  return `(helper function ${fn.name})`;
} else {
  return '(anonymous helper function)';
}
```

Is there an existing format/convention from the helper manager that we can steal/align with?

Reply:
> could do something like:

sounds good to me!

> Is there an existing format/convention from the helper manager that we can steal/align with?

I didn't see one as I was browsing around